### PR TITLE
added doxygen comments to the contract and member functions for alu.sol and vmmemory.sol

### PR DESCRIPTION
**Pull request by bloodstalker**
just like the title says.
if everybody's happy with the style, i can just do this for all our solidity files.

### CODE REVIEW

File: alu.sol [hunk 1]
```diff
@@ -2,7 +2,22 @@ pragma solidity ^0.4.15;
...
+*/
 contract ALU is VMMemory {
+  /**
+    * @dev handles the ALU operations of the WASM machine i.e. the WASM instructions are implemented and run here
+  *
+    * @param hint
+    * @param r1 register one
+    * @param r2 register two
+    * @param r3 register three
+    * @param ireg the other register
```
Comment (mrsmkl) [inline alu.sol]: ireg is the register that has the immediate value

File: vmmemory.sol [hunk 2]
```diff
@@ -24,18 +56,44 @@ contract VMMemory {
...
         a = loadN(mem, 0, 8);
         b = loadN(mem, 8, 8);
     }
     
+    /**
+      * @dev returns the sizes of different WASM types in bytes
+    *
+      * @param ty uint value represeting a WASM type
+    *
+      * @return the bumber of bytes ty has
```
Comment (mrsmkl) [inline vmmemory.sol]: number

File: alu.sol [hunk 3]
```diff
@@ -2,7 +2,22 @@ pragma solidity ^0.4.15;
...
 
+/**
+  * @title the ALU for the solidity interpreter.
+  * @author Sami Mäkelä
+*/
 contract ALU is VMMemory {
+  /**
+    * @dev handles the ALU operations of the WASM machine i.e. the WASM instructions are implemented and run here
+  *
+    * @param hint
```
Comment (mrsmkl) [inline alu.sol]: Must have description. it is the alu instruction code

**Comment by bloodstalker** [comment]
I'll fix those. Add doxygen comments to the rest of the solidity files. I'll also squish the commits at some point.

**Comment by bloodstalker** [comment]
closing due to deprecation. new one is #70.

**Pull request closed without merging**